src/CMakeLists.txt

```diff
@@ -300,6 +300,7 @@ endif()
 # tcmalloc heap profiler
 set(heap_profiler_files ${TCMALLOC_srcs})
 add_library(heap_profiler_objs OBJECT ${heap_profiler_files})
+add_dependencies(heap_profiler_objs Boost::boost)
```

Comment:
these deps are how we trigger the external project, and guarantee that the boost headers are available?

if we add a new target that doesn't add this dependency, does it fail somehow? or would it silently pull in the boost system headers?

this seems error-prone, but i'm not sure how to reconcile this with the need for `include_directories(BEFORE SYSTEM ${Boost_INCLUDE_DIRS})` to override the system headers